bfvmm/src/vmcs/src/vmcs_intel_x64_check_controls.cpp

```diff
@@ -637,63 +615,57 @@ vmcs_intel_x64::check_control_event_injection_delivery_ec_checks()
             break;
 
         default:
-            throw std::logic_error("vector must indicate exception that would nomrally "
-                                   "deliver an error code if deliver error code bit is set");
+            if (deliver_error_code_bit::is_set())
+                throw std::logic_error("vector must indicate exception that would normally "
+                                       "deliver an error code if deliver_error_code_bit is set");
     }
+
+    if (!deliver_error_code_bit::is_set())
+        throw std::logic_error("deliver_error_code_bit must be 1");
 }
 
 void
 vmcs_intel_x64::check_control_event_injection_reserved_bits_checks()
 {
-    auto interrupt_info_field =
-        vm::read(VMCS_VM_ENTRY_INTERRUPTION_INFORMATION_FIELD);
-
-    if ((interrupt_info_field & VM_INTERRUPT_INFORMATION_VALID) == 0)
+    if (!vmcs::vm_entry_interruption_information_field::valid_bit::is_set())
         return;
 
-    if ((interrupt_info_field & 0x000000007FFFF000) != 0)
+    if (vmcs::vm_entry_interruption_information_field::reserved::get() != 0)
         throw std::logic_error("reserved bits of the interrupt info field must be 0");
 }
 
 void
 vmcs_intel_x64::check_control_event_injection_ec_checks()
 {
-    auto interrupt_info_field =
-        vm::read(VMCS_VM_ENTRY_INTERRUPTION_INFORMATION_FIELD);
+    using namespace vmcs::vm_entry_interruption_information_field;
 
-    auto exception_error_code =
-        vm::read(VMCS_VM_ENTRY_EXCEPTION_ERROR_CODE);
-
-    if ((interrupt_info_field & VM_INTERRUPT_INFORMATION_VALID) == 0)
+    if (!valid_bit::is_set())
         return;
 
-    if ((interrupt_info_field & VM_INTERRUPT_INFORMATION_DELIVERY_ERROR) == 0)
+    if (!deliver_error_code_bit::is_set())
         return;
 
-    if ((exception_error_code & 0x00000000FFFF8000) != 0)
+    if ((vmcs::vm_entry_exception_error_code::get() & 0x00000000FFFF8000UL) != 0)
         throw std::logic_error("bits 31:15 of the exception error code field must be 0 "
                                "if deliver error code bit is set in the interrupt info field");
 }
 
 void
 vmcs_intel_x64::check_control_event_injection_instr_length_checks()
```

Comment:
The way you coded this is awesome. I never thought to use the `using` feature to qualify a type, which is pretty badass. Great job here!!!